File: Raspberry Pi App/barcode_scan_app/lib/data_classes/product_info.dart
class ProductInfo {
  ProductInfo({required this.eancode, required this.name});

  final String eancode;
  final String name;
  bool finishedEditing = false;
  bool isExpired = false;
  int quantity = 1;
  DateTime? expiryDate;

  Map<String, dynamic> toFirestore() {
    return {
      'eancode': eancode,
      'name': name,
      'finishedEditing': finishedEditing,
      'quantity': quantity,
      'expiryDate': expiryDate,
      'isExpired': isExpired
    };
  }
}

File: Raspberry Pi App/Python_App/product_info.py
from dataclasses import dataclass, field
from typing import Optional, Dict
from datetime import datetime

@dataclass
class ProductInfo:
    eancode: str
    name: str
    finishedEditing: bool = False
    isExpired: bool = False
    imageURL: str = ''
    categories: list = field(default_factory=list)
    quantity: int = 1
    expiryDate: Optional[datetime] = None

    def toFirestore(self) -> Dict[str, any]:
        return {
            'eancode': self.eancode,
            'name': self.name,
            'finishedEditing': self.finishedEditing,
            'isExpired': self.isExpired,
            'categories': self.categories,
            'imageURL': self.imageURL,
            'quantity': self.quantity,
            'expiryDate': self.expiryDate
        }